sectionsUtilityTool/src/main/java/edu/umich/ctools/sectionsUtilityTool/SectionsUtilityToolServlet.java

```diff
@@ -15,93 +15,131 @@
+	private String canvasToken;
+	private String canvasURL;
+	
+	
+	public void init() throws ServletException {
+		M_log.debug("init(): Called");
+		getCanvasCredentials();
+		
+	}
 	
 	protected void doGet(HttpServletRequest request,HttpServletResponse response) throws IOException {
 		M_log.debug("doGet: Called");
-		HttpClient client = new DefaultHttpClient();
-		String targetURI= "https://umich.test.instructure.com";
+		canvasRestApiCall(request, response);
+	}
+	
+	protected void doPost(HttpServletRequest request,HttpServletResponse response) throws IOException {
+		M_log.debug("doPOST: Called");
+		canvasRestApiCall(request, response);
+		
+	}
+	protected void doPut(HttpServletRequest request,HttpServletResponse response) throws IOException {
+		M_log.debug("doPut: Called");
+		canvasRestApiCall(request, response);
+		
+	}
+	
+	
+
+	private void canvasRestApiCall(HttpServletRequest request,
+			HttpServletResponse response) throws IOException {
+		request.setCharacterEncoding("UTF-8");
+		M_log.debug("canvasRESTAPICall(): called");
+		if(canvasProperties!=null) {
+			canvasToken = canvasProperties.getProperty(PROPERTY_CANVAS_ADMIN);
+			canvasURL = canvasProperties.getProperty(PROPERTY_CANVAS_URL);
+		}
+		else {
+			response.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+			PrintWriter out = response.getWriter();
+			out.print("Problem loading property file");
+			out.flush();
+			M_log.error("Failed to load system properties(sectionsToolProps.properties) for SectionsTool");
+			return;
+		}
 		String queryString = request.getQueryString();
-		String servletPath = request.getServletPath();
 		String pathInfo = request.getPathInfo();
 		String url=null;
 		if(pathInfo!=null&&(pathInfo.contains("courses")||pathInfo.contains("terms")||pathInfo.contains("crosslist"))) {
 		if(queryString!=null) {
-		url= targetURI+pathInfo+"?"+queryString;
-		 //url= servletPath+"?"+queryString;
-		 }
-		else {
-			url=targetURI+pathInfo;
-			//url=servletPath;
+		url= canvasURL+pathInfo+"?"+queryString;
+		 }else {
+			url=canvasURL+pathInfo;
 		}
-		System.out.println("URL: "+url);
-		HttpGet clientRequest = new HttpGet(url);
+		HttpUriRequest clientRequest = null;
+		if(request.getMethod().equals(GET)) {
+			clientRequest = new HttpGet(url);
+		}else if (request.getMethod().equals(POST)) {
+			clientRequest = new HttpPost(url);
+		}else if(request.getMethod().equals(PUT)) {
+			clientRequest=new HttpPut(url);
+		}
+		HttpClient client = new DefaultHttpClient();
 		final ArrayList<NameValuePair> nameValues = new ArrayList<NameValuePair>();
-	    nameValues.add(new BasicNameValuePair("Authorization", "Bearer"+ " " +getCanvasAdminToken()));
-	    nameValues.add(new BasicNameValuePair("content-type", "application/json"));
-	    for (final NameValuePair h : nameValues)
-	    {
-	        clientRequest.addHeader(h.getName(), h.getValue());
-	    }
+		nameValues.add(new BasicNameValuePair("Authorization", "Bearer"+ " " +canvasToken));
```

Comment:
A comment here would be good, since we're using a special admin account to handle all canvas requests for any user